### PR TITLE
React/pt 68/login screen flow

### PR DESCRIPTION
**Pull request by eduardb**
TL;DR Proper screen  flow on app startup: splash screen -> user logged in ? tweets timeline : login screen

We decided to have a splash screen as the entry point in the app because we have to load asynchronously the stored user credentials, if any.  If there are no stored credentials, we go to the login screen, else to the user tweets timeline (which is broken atm until we introduce the oauth stuff there as well).

We also decided to move the components related to debug in a `debug` subfolder.

The debug screen is inaccessible atm, as soon as we introduce a toolbar / nav bar, we will add a menu item to access it again :D

| iOS  | Android |
| --- | --- |
| screenshot-ios | screenshot-android | 

Paired with @gbasile

### CODE REVIEW

File: ReactNative/ReactTwitter/core/views/splash-screen.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+  },
+
+  componentDidMount () {
+    let authService = this.state.authenticationService
+
+    authService.loadDataFromDisk()
+      .then(() => {
+        return new Promise((resolve, reject) => {
+          setTimeout(resolve, 1100)
```
Comment (takecare) [inline ReactNative/ReactTwitter/core/views/splash-screen.js]: why 1100?

Comment (eduardb) [inline ReactNative/ReactTwitter/core/views/splash-screen.js]: it was carefully chosen for a good UX impact 👅 
I will extract it to a constant

File: ReactNative/ReactTwitter/core/views/login-screen.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        .catch(console.warn)
+  },
+
+  _browserAuthenticationWithToken (oauthToken) {
+    this.state.facade.listenForDeepLinking().then((uri) => {
+      let parsedURI = OauthHelper.getOauthTokenAndVerifierFromURLCallback(uri)
+      this.state.facade.stopListeningForDeepLinking()
+
+      if (parsedURI.oauth_token == null || parsedURI.oauth_token.length === 0) {
+        // user canceled login
```
Comment (xrigau) [inline ReactNative/ReactTwitter/core/views/login-screen.js]: this can be replaced with `if (userCancelledLogin())` 👅

File: ReactNative/ReactTwitter/core/views/login-screen.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            } else {
+              console.log('Something went wrong :o')
+            }
+          })
+        })
+    })
+    Linking.openURL('https://api.twitter.com/oauth/authenticate?oauth_token=' + oauthToken)
+  },
+
+  _pushTweetsList () {
```
Comment (xrigau) [inline ReactNative/ReactTwitter/core/views/login-screen.js]: is the leading `_` a convention?

Comment (eduardb) [inline ReactNative/ReactTwitter/core/views/login-screen.js]: Yes, it is :P

File: ReactNative/ReactTwitter/core/views/login-screen.js (hunk 2 above)
Comment (eduardb) [inline ReactNative/ReactTwitter/core/views/login-screen.js]: I will take into consideration your suggestion for a future PR

File: ReactNative/ReactTwitter/core/views/login-screen.js (hunk 3 above)
Comment (xrigau) [inline ReactNative/ReactTwitter/core/views/login-screen.js]: 👍 same with the space before the `(` 😷

Comment (eduardb) [inline ReactNative/ReactTwitter/core/views/login-screen.js]: The linter will complain if we don't put the space. Probably that we could alter the rules, but it's a bit too late for that probably.

Comment (gbasile) [inline ReactNative/ReactTwitter/core/views/login-screen.js]: We are using the default preset for the linter, I would imagine this is pretty standard

`{
    "extends": ["standard", "plugin:react/recommended"],
    "plugins": [
        "standard",
        "react"
    ]
}`
